.github/workflows/ci.yml

```diff
@@ -85,30 +61,19 @@ on:
       caching_after_step_command:
         type: string
         default: ""
-      docker_before_step_command:
-        type: string
-        default: ""
-      docker_after_step_command:
+      technology_based_scans_before_step_command:  
         type: string
         default: ""
-      lint_before_step_command:
+      technology_based_scans_after_step_command:  
         type: string
-        default: ""
-      lint_after_step_command:
-        type: string
-        default: ""
-      build_before_step_command:
-        type: string
-        default: ""
-      build_after_step_command:
-        type: string
-        default: ""
+        default: ""        
 
 jobs:
   caching:
     name: Cache node modules
     runs-on: ubuntu-latest
     permissions: write-all
+    if: ${{!contains(github.actor, 'github-actions[bot]')  && (github.event_name != 'issue_comment') && (github.actor != 'dependabot[bot]') }}
```

Comment:
github-actions[bot] check is happening twice in this `if` statement itself. Should be checked only once.
And combine all the checks into contains() func itself.